app/services/remove_service.rb

```diff
@@ -1,35 +1,51 @@
 # frozen_string_literal: true
 
-class RemoveService
-  def call(edition, removal)
+class RemoveService < ApplicationService
+  def initialize(edition, removal)
+    @edition = edition
+    @removal = removal
+  end
+
+  def call
     Document.transaction do
       edition.document.lock!
-      check_removeable(edition)
-
-      GdsApi.publishing_api_v2.unpublish(
-        edition.content_id,
-        type: removal.redirect? ? "redirect" : "gone",
-        explanation: removal.explanatory_note,
-        alternative_path: removal.alternative_path,
-        locale: edition.locale,
-      )
-
-      edition.assign_status(:removed, nil, status_details: removal)
-      edition.save!
-
-      TimelineEntry.create_for_status_change(
-        entry_type: :removed,
-        status: edition.status,
-        details: removal,
-      )
+      check_removeable
+      unpublish_edition
+      update_edition_status
+      create_timeline_entry
     end
 
-    delete_assets(edition.assets)
+    delete_assets
```

Comment:
Although I'm not sure this commit was the right place to do this refactoring I do like how this service is wrote and hope this can be the conventional way our call methods look 👍

Reply:
Agreed. If we had more time, I'd have preferred to make these changes as a separate commit/PR, since this seems to be the main opportunity to align with the approach we have elsewhere.